### PR TITLE
The Host value should be written to all rejected routes

### LINKED ISSUE #7350: Route ingress that has failed admission is missing the host details
Example route details...

```
> oc get route and-failure -o yaml
apiVersion: v1
kind: Route
metadata:
  creationTimestamp: 2016-02-16T00:09:21Z
  labels:
    app: nodejs
  name: and-failure
  namespace: example
  resourceVersion: "3489"
  selfLink: /oapi/v1/namespaces/example/routes/and-failure
  uid: 87bae5ea-d441-11e5-bc18-54ee75107c12
spec:
  host: www.cdrom.com
  port:
    targetPort: 8080-tcp
  to:
    kind: Service
    name: nodejs
status:
  ingress:
  - conditions:
    - lastTransitionTime: 2016-02-16T00:09:21Z
      message: route anotherone already exposes www.cdrom.com and is older
      reason: HostAlreadyClaimed
      status: "False"
      type: Admitted
    routerName: router
```

### PR DESCRIPTION
**Pull request by smarterclayton**
Test cases were broken and not catching this error.

Fixes #7350

@jwforres @ramr

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by jwforres** [comment]
main change looks good, i'll let @ramr review the testcase

File: pkg/router/controller/unique_host.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 		} else {
 			if oldest.CreationTimestamp.Before(route.CreationTimestamp) {
 				glog.V(4).Infof("Route %s cannot take %s from %s", routeName, host, routeNameKey(oldest))
-				err := fmt.Errorf("another route holds %s and is older than %s", host, route.Name)
+				err := fmt.Errorf("a route in another namespace holds %s and is older than %s", host, route.Name)
```
Comment (marun) [inline pkg/router/controller/unique_host.go]: :+1:

**Comment by smarterclayton** [comment]
Or @pweil-

**Comment by pweil-** [comment]
LGTM

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**